### PR TITLE
Add CLI option to save CLI report summary to disk as JSON/CSV

### LINKED ISSUE #525: Add summary report CLI for a scan
## Specify type:
- Enhancement

## Description:
A summary report utilizes the `/api/v1/reports/?fact_collection_id` endpoint.

Interaction with the CLI should be as follows:

`qpc report summary --id <scan_id> (--json | --csv) --output-file <path>`
Summary report outputs to a single file of the specified format.

The CLI code can extract the `fact_collection_id` from the scan, if available, and return the resulting data. Supporting output of JSON and CSV.  If the user selects CSV then any multi-dimensional data like a list will be converted into a string separated with semi-colons `;`, dictionaries will be key `:` value separated by semi-colons.

### LINKED ISSUE #592: Support CSV (text/csv) from /api/v1/reports/?fact_collection_id endpoint.
## Specify type:
- Enhancement

## Description:
Support Accept type text/csv from `/api/v1/reports/?fact_collection_id` endpoint.

Summary report outputs to a single file of the specified format.

The CLI code can extract the `fact_collection_id` from the scan, if available, and return the resulting data. Supporting output of JSON and CSV.  If the user selects CSV then any multi-dimensional data like a list will be converted into a string separated with semi-colons `;`, dictionaries will be key `:` value separated by semi-colons.

This issue enables #525  to be completed.

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #525 
Closes #592

### CODE REVIEW

File: qpc/messages.py [hunk 1]
```diff
@@ -116,6 +116,17 @@
 SCAN_RESTARTED = 'Scan "%s" restarted.'
 SCAN_CANCELED = 'Scan "%s" canceled.'
 
+REPORT_SUMMARY_ID_HELP = 'Fact collection identifier.'
```
Comment (chambridge) [inline qpc/messages.py]: Supposed to take the Scan ID and pull the fact collection ID from there

File: qpc/messages.py [hunk 2]
```diff
@@ -116,6 +116,17 @@
...
+REPORT_SUMMARY_OUTPUT_JSON_HELP = 'Output as a JSON file.'
+REPORT_SUMMARY_OUTPUT_CSV_HELP = 'Output as a CSV file.'
+REPORT_SUMMARY_PATH_HELP = 'Output file location.'
+REPORT_SUMMARY_FC_DOES_NOT_EXIST = \
```
Comment (chambridge) [inline qpc/messages.py]: Should be scan id

File: qpc/messages.py [hunk 3]
```diff
@@ -134,6 +145,8 @@
                 'status of the server.'
 
 READ_FILE_ERROR = 'Error reading from %s: %s.'
+WRITE_FILE_ERROR = 'Error writing to %s: %s.'
+FILE_SUCCESSFULLY_WRITTEN = 'File successfully written to %s.'
```
Comment (chambridge) [inline qpc/messages.py]: Report written successfully.

File: qpc/report/commands.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2017 Red Hat, Inc.
```
Comment (chambridge) [inline qpc/report/commands.py]: 2018

File: qpc/report/summary.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2017 Red Hat, Inc.
```
Comment (chambridge) [inline qpc/report/summary.py]: 2018

File: qpc/report/summary.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        group.add_argument('--csv', dest='output_csv', action='store_true',
+                           help=_(messages.REPORT_SUMMARY_OUTPUT_CSV_HELP))
+
+        self.parser.add_argument('--output-file', dest='path', metavar='PATH',
+                                 help=_(messages.REPORT_SUMMARY_PATH_HELP),
+                                 required=True)
+
+    def _validate_args(self):
+        CliCommand._validate_args(self)
+        if self.args.id:
```
Comment (chambridge) [inline qpc/report/summary.py]: Need to take Scan ID and retrieve fact collection. This is a simpler flow for users.

File: qpc/messages.py (hunk 1 above)
Comment (kholdaway) [reply qpc/messages.py]: Ok.  Fixed.

File: qpc/messages.py (hunk 2 above)
Comment (kholdaway) [reply qpc/messages.py]: Fixed

File: qpc/messages.py (hunk 3 above)
Comment (kholdaway) [reply qpc/messages.py]: Changed

File: qpc/report/commands.py (hunk 4 above)
Comment (kholdaway) [reply qpc/report/commands.py]: Updated

File: qpc/report/summary.py (hunk 5 above)
Comment (kholdaway) [reply qpc/report/summary.py]: Updated

**Pull request merged**